placement/algo/sharded_test.go

```diff
@@ -1174,7 +1203,7 @@ func validateDistribution(t *testing.T, p placement.Placement, expectPeakOverAvg
 				testCase, i.ID(), instanceOverTarget, load, targetLoad))
 		}
 	}
-	assert.Equal(t, total, p.ReplicaFactor()*p.NumShards(), fmt.Sprintf("Wrong total shards: expecting %v, but got %v", p.ReplicaFactor()*p.NumShards(), total))
+	assert.Equal(t, p.ReplicaFactor()*p.NumShards(), total, fmt.Sprintf("Wrong total shards: expecting %v, but got %v", p.ReplicaFactor()*p.NumShards(), total))
```

Comment:
given that we are requiring that all shards be marked as available, as indicated by the use of must in the function, perhaps we want to use `require` instead of `assert` here? With `assert` the tests will continue even if condition is not true, but with `require` they will stop immediately.

Reply:
we use assert instead of require in many places in this test so we can capture more errors in one test, I felt it helped investigating. Can change some NoError checks to require